inc/common/admin-bar.php

```diff
@@ -12,299 +12,300 @@
-		)
+		]
 	);
 
-	if ( rocket_valid_key() ) {
+	/**
+	 * Purge Cache.
+	 */
+	$action = 'purge_cache';
 
-		if ( rocket_is_plugin_active( 'sitepress-multilingual-cms/sitepress.php' ) ) {
+	if ( rocket_valid_key() ) {
+		$i18n_plugin = rocket_has_i18n();
 
-			// Purge All.
+		if ( $i18n_plugin ) {
+			// Parent.
 			$wp_admin_bar->add_menu(
-				array(
+				[
 					'parent' => 'wp-rocket',
 					'id'     => 'purge-all',
 					'title'  => __( 'Clear cache', 'rocket' ),
 					'href'   => '#',
-				)
+				]
 			);
 
-			$langlinks = get_rocket_wpml_langs_for_admin_bar();
-			if ( $langlinks ) {
+			// Add submenu for each active langs.
+			if ( 'wpml' === $i18n_plugin ) {
+				$langlinks = get_rocket_wpml_langs_for_admin_bar();
+			} elseif ( 'qtranslate' === $i18n_plugin ) {
+				$langlinks = get_rocket_qtranslate_langs_for_admin_bar();
+			} elseif ( 'qtranslate-x' === $i18n_plugin ) {
+				$langlinks = get_rocket_qtranslate_langs_for_admin_bar( 'x' );
+			} elseif ( 'polylang' === $i18n_plugin ) {
+				$langlinks = get_rocket_polylang_langs_for_admin_bar();
+			} else {
+				$langlinks = [];
+			}
 
+			if ( $langlinks ) {
 				foreach ( $langlinks as $lang ) {
 					$wp_admin_bar->add_menu(
-						array(
+						[
 							'parent' => 'purge-all',
 							'id'     => 'purge-all-' . $lang['code'],
 							'title'  => $lang['flag'] . '&nbsp;' . $lang['anchor'],
 							'href'   => wp_nonce_url( admin_url( 'admin-post.php?action=' . $action . '&type=all&lang=' . $lang['code'] . $referer ), $action . '_all' ),
-						)
+						]
 					);
 				}
-			}
-		} elseif ( rocket_is_plugin_active( 'qtranslate/qtranslate.php' ) || rocket_is_plugin_active( 'qtranslate-x/qtranslate.php' ) || rocket_is_plugin_active( 'polylang/polylang.php' ) || rocket_is_plugin_active( 'polylang-pro/polylang.php' ) ) {
 
-			// Purge All.
-			$wp_admin_bar->add_menu(
-				array(
-					'parent' => 'wp-rocket',
-					'id'     => 'purge-all',
-					'title'  => __( 'Clear cache', 'rocket' ),
-					'href'   => '#',
-				)
-			);
-
-			// Add submenu for each active langs.
-			if ( rocket_is_plugin_active( 'qtranslate/qtranslate.php' ) ) {
-				$langlinks = get_rocket_qtranslate_langs_for_admin_bar();
-			} elseif ( rocket_is_plugin_active( 'qtranslate-x/qtranslate.php' ) ) {
-				$langlinks = get_rocket_qtranslate_langs_for_admin_bar( 'x' );
-			} elseif ( rocket_is_plugin_active( 'polylang/polylang.php' ) || rocket_is_plugin_active( 'polylang-pro/polylang.php' ) ) {
-				$langlinks = get_rocket_polylang_langs_for_admin_bar();
-			}
-
-			foreach ( $langlinks as $lang ) {
-				$wp_admin_bar->add_menu(
-					array(
-						'parent' => 'purge-all',
-						'id'     => 'purge-all-' . $lang['code'],
-						'title'  => $lang['flag'] . '&nbsp;' . $lang['anchor'],
-						'href'   => wp_nonce_url( admin_url( 'admin-post.php?action=' . $action . '&type=all&lang=' . $lang['code'] . $referer ), $action . '_all' ),
-					)
-				);
+				if ( 'wpml' !== $i18n_plugin ) {
```

Comment:
this should be before the previous conditional, to have the "All" link first in the list

Reply:
It was at the end of list until now, do you want me to change it?